packages/cursorless-engine/src/languages/TreeSitterQuery/queryPredicateOperators.ts

```diff
@@ -116,6 +127,36 @@ class ChildRange extends QueryPredicateOperator<ChildRange> {
   }
 }
 
+/**
+ * A predicate operator that modifies the range of the match to shrink to regex
+ * match.  For example, `(#shrink-to-match! @foo "\\S+")` will modify the range
+ * of the `@foo` capture to exclude whitespace.
+ */
+class ShrinkToMatch extends QueryPredicateOperator<ShrinkToMatch> {
+  name = "shrink-to-match!" as const;
+  schema = z.tuple([q.node, q.string]);
+
+  run(nodeInfo: MutableQueryCapture, pattern: string) {
+    const {
+      range,
+      node: { text },
+    } = nodeInfo;
+
+    const match = text.match(new RegExp(pattern));
+
+    if (match?.index == null) {
+      throw Error(`No match for pattern '${pattern}'`);
+    }
+
+    nodeInfo.range = new Range(
```

Comment:
I just realize why I used to have an if statement that throw and error on multiple lines. Since we don't have access to the editor we can't convert in decease to positions. Just translating position character only works if it's on the same line.

Reply:
Ah right. Should be fairly straightforward to add an editor field I'd think no? Would be a bit surprising if things fail when you break a line 😅